File: test_gemini.js
const fs = require('fs');
const jsdom = require('jsdom');
const { JSDOM } = jsdom;

const html = fs.readFileSync('sidebar/gemini.html', 'utf8');
const dom = new JSDOM(html, { url: "https://gemini.google.com/" });
const document = dom.window.document;

const root = document.querySelector('bard-sidenav');
console.log("Root found:", !!root);
if (root) {
  const links = root.querySelectorAll('a[href^="/app/"], a[href^="https://gemini.google.com/app/"]');
  console.log("Links inside root:", links.length);
}

const allLinks = document.querySelectorAll('a[href^="/app/"], a[href^="https://gemini.google.com/app/"]');
console.log("All links in document:", allLinks.length);

File: test_gemini.py
import re
from html.parser import HTMLParser

class MyHTMLParser(HTMLParser):
    def __init__(self):
        super().__init__()
        self.in_sidenav = 0
        self.links_in_sidenav = set()
        self.links_outside = set()

    def handle_starttag(self, tag, attrs):
        if tag == 'bard-sidenav':
            self.in_sidenav += 1
        elif tag == 'a':
            href = dict(attrs).get('href', '')
            if '/app/' in href:
                match = re.search(r'/app/([a-zA-Z0-9]+)', href)
                if match:
                    path = match.group(0)
                    if self.in_sidenav > 0:
                        self.links_in_sidenav.add(path)
                    else:
                        self.links_outside.add(path)

    def handle_endtag(self, tag):
        if tag == 'bard-sidenav':
            self.in_sidenav -= 1

parser = MyHTMLParser()
with open('sidebar/gemini.html', 'r', encoding='utf-8') as f:
    parser.feed(f.read())

print("In sidenav:", len(parser.links_in_sidenav))
print("Outside sidenav:", len(parser.links_outside))
